pkg/config/nodetreemodel/read_config_file.go

```diff
@@ -134,11 +134,16 @@ func loadYamlInto(dest InnerNode, source model.Source, inData map[string]interfa
 			warnings = append(warnings, fmt.Errorf("unknown key from YAML: %s", currPath))
 			if !allowDynamicSchema {
 				continue
-			} else if isScalar(value) || isSlice(value) {
-				schemaChild = newLeafNode(value, model.SourceSchema)
-			} else {
-				schemaChild = newInnerNode(make(map[string]Node))
 			}
+
+			// if the key is not defined in the schema, we can still add it to the destination
+			if value == nil || isScalar(value) || isSlice(value) {
```

Comment:
Do you know of a situation where we end up with `value == nil` here? Is there a unit test covering that case?